--- target/expected_output/basic/akka/rk/buh/is/it/A.java
+++ target/expected_output/basic/akka/rk/buh/is/it/A.java
@@ -48,13 +48,13 @@
     public  class NonStatic {
       public   NonStatic ()  { throw new RuntimeException(); }
     }
+    public  akka.rk.buh.is.it.A.D$.E$ E ()  { throw new RuntimeException(); }
     public   D$ ()  { throw new RuntimeException(); }
     /**
      * def A.D.math
      * @return (undocumented)
      */
     public  long math ()  { throw new RuntimeException(); }
-    public  akka.rk.buh.is.it.A.D$.E$ E ()  { throw new RuntimeException(); }
   }
   /**
    * class A.B
@@ -94,12 +94,12 @@
     public  class C1$ {
       public   C1$ ()  { throw new RuntimeException(); }
     }
-    public   C1 ()  { throw new RuntimeException(); }
     /**
      * Accessor for nested Scala object
      * @return (undocumented)
      */
     public  akka.rk.buh.is.it.A.C1.C1$ C1 ()  { throw new RuntimeException(); }
+    public   C1 ()  { throw new RuntimeException(); }
   }
   /**
    * object C1
@@ -147,6 +147,11 @@
   static public  java.lang.String stattic ()  { throw new RuntimeException(); }
   static public  java.lang.Object x ()  { throw new RuntimeException(); }
   /**
+   * Accessor for nested Scala object
+   * @return (undocumented)
+   */
+  public  akka.rk.buh.is.it.A.D$ D ()  { throw new RuntimeException(); }
+  /**
    * varargs
    * @param s (undocumented)
    * @return (undocumented)
@@ -219,9 +224,4 @@
    */
   public  int testthrows () throws java.lang.IllegalArgumentException, java.lang.NullPointerException { throw new RuntimeException(); }
   public  scala.runtime.Null$ getNull ()  { throw new RuntimeException(); }
-  /**
-   * Accessor for nested Scala object
-   * @return (undocumented)
-   */
-  public  akka.rk.buh.is.it.A.D$ D ()  { throw new RuntimeException(); }
 }
