--- target/expected_output/basic/akka/rk/buh/is/it/A.java
+++ target/expected_output/basic/akka/rk/buh/is/it/A.java
@@ -48,13 +48,13 @@
     public  class NonStatic {
       public   NonStatic ()  { throw new RuntimeException(); }
     }
+    public  akka.rk.buh.is.it.A.D$.E$ E ()  { throw new RuntimeException(); }
     public   D$ ()  { throw new RuntimeException(); }
     /**
      * def A.D.math
      * @return (undocumented)
      */
     public  long math ()  { throw new RuntimeException(); }
-    public  akka.rk.buh.is.it.A.D$.E$ E ()  { throw new RuntimeException(); }
   }
   /**
    * class A.B
@@ -76,7 +76,7 @@
      * def b(args: java.lang.String*): Unit
      * @param args (undocumented)
      */
-    public  void b (scala.collection.Seq<java.lang.String> args)  { throw new RuntimeException(); }
+    public  void b (scala.collection.immutable.Seq<java.lang.String> args)  { throw new RuntimeException(); }
     public  java.lang.String d (java.lang.String a, akka.rk.buh.is.it.X b)  { throw new RuntimeException(); }
   }
   public  class C implements akka.rk.buh.is.it.X {
@@ -94,12 +94,12 @@
     public  class C1$ {
       public   C1$ ()  { throw new RuntimeException(); }
     }
-    public   C1 ()  { throw new RuntimeException(); }
     /**
      * Accessor for nested Scala object
      * @return (undocumented)
      */
     public  akka.rk.buh.is.it.A.C1.C1$ C1 ()  { throw new RuntimeException(); }
+    public   C1 ()  { throw new RuntimeException(); }
   }
   /**
    * object C1
@@ -147,6 +147,11 @@
   static public  java.lang.String stattic ()  { throw new RuntimeException(); }
   static public  java.lang.Object x ()  { throw new RuntimeException(); }
   /**
+   * Accessor for nested Scala object
+   * @return (undocumented)
+   */
+  public  akka.rk.buh.is.it.A.D$ D ()  { throw new RuntimeException(); }
+  /**
    * varargs
    * @param s (undocumented)
    * @return (undocumented)
@@ -212,16 +217,11 @@
    * @param s (undocumented)
    * @return (undocumented)
    */
-  public  int hello (scala.collection.Seq<java.lang.String> s)  { throw new RuntimeException(); }
+  public  int hello (scala.collection.immutable.Seq<java.lang.String> s)  { throw new RuntimeException(); }
   /**
    * throws
    * @return (undocumented)
    */
   public  int testthrows () throws java.lang.IllegalArgumentException, java.lang.NullPointerException { throw new RuntimeException(); }
   public  scala.runtime.Null$ getNull ()  { throw new RuntimeException(); }
-  /**
-   * Accessor for nested Scala object
-   * @return (undocumented)
-   */
-  public  akka.rk.buh.is.it.A.D$ D ()  { throw new RuntimeException(); }
 }
