--- target/expected_output/basic/akka/rk/buh/is/it/A.java
+++ target/expected_output/basic/akka/rk/buh/is/it/A.java
@@ -48,13 +48,13 @@
     public  class NonStatic {
       public   NonStatic ()  { throw new RuntimeException(); }
     }
+    public  akka.rk.buh.is.it.A.D$.E$ E ()  { throw new RuntimeException(); }
     public   D$ ()  { throw new RuntimeException(); }
     /**
      * def A.D.math
      * @return (undocumented)
      */
     public  long math ()  { throw new RuntimeException(); }
-    public  akka.rk.buh.is.it.A.D$.E$ E ()  { throw new RuntimeException(); }
   }
   /**
    * class A.B
@@ -76,7 +76,7 @@
      * def b(args: java.lang.String*): Unit
      * @param args (undocumented)
      */
-    public  void b (scala.collection.Seq<java.lang.String> args)  { throw new RuntimeException(); }
+    public  void b (scala.collection.immutable.Seq<java.lang.String> args)  { throw new RuntimeException(); }
     public  java.lang.String d (java.lang.String a, akka.rk.buh.is.it.X b)  { throw new RuntimeException(); }
   }
   public  class C implements akka.rk.buh.is.it.X {
@@ -94,12 +94,12 @@
     public  class C1$ {
       public   C1$ ()  { throw new RuntimeException(); }
     }
-    public   C1 ()  { throw new RuntimeException(); }
     /**
      * Accessor for nested Scala object
      * @return (undocumented)
      */
     public  akka.rk.buh.is.it.A.C1.C1$ C1 ()  { throw new RuntimeException(); }
+    public   C1 ()  { throw new RuntimeException(); }
   }
   /**
    * object C1
@@ -147,6 +147,11 @@
   static public  java.lang.String stattic ()  { throw new RuntimeException(); }
   static public  java.lang.Object x ()  { throw new RuntimeException(); }
   /**
+   * Accessor for nested Scala object
+   * @return (undocumented)
+   */
+  public  akka.rk.buh.is.it.A.D$ D ()  { throw new RuntimeException(); }
+  /**
    * varargs
    * @param s (undocumented)
    * @return (undocumented)
@@ -212,16 +217,11 @@
    * @param s (undocumented)
    * @return (undocumented)
    */
-  public  int hello (scala.collection.Seq<java.lang.String> s)  { throw new RuntimeException(); }
+  public  int hello (scala.collection.immutable.Seq<java.lang.String> s)  { throw new RuntimeException(); }
   /**
    * throws
    * @return (undocumented)
    */
   public  int testthrows () throws java.lang.IllegalArgumentException, java.lang.NullPointerException { throw new RuntimeException(); }
   public  scala.runtime.Null$ getNull ()  { throw new RuntimeException(); }
-  /**
-   * Accessor for nested Scala object
-   * @return (undocumented)
-   */
-  public  akka.rk.buh.is.it.A.D$ D ()  { throw new RuntimeException(); }
 }
diff --git a/plugin/src/test/resources/expected_output/basic/akka/rk/buh/is/it/Blarb.java b/plugin/src/test/resources/expected_output/basic/akka/rk/buh/is/it/Blarb.java
index b0dd497..4950950 100644
--- target/expected_output/basic/akka/rk/buh/is/it/Blarb.java
+++ target/expected_output/basic/akka/rk/buh/is/it/Blarb.java
@@ -1,52 +1,51 @@
 package akka.rk.buh.is.it;
-// no position
 /**
  * I am an object. I have no class.
  */
 public  class Blarb {
-  // no position
   static public  class A$ implements akka.rk.buh.is.it.Blarb.Fuz, scala.Product, scala.Serializable {
     /**
      * Static reference to the singleton instance of this Scala object.
      */
     public static final A$ MODULE$ = null;
-    public   A$ () { throw new RuntimeException(); }
-    public  java.lang.String productPrefix () { throw new RuntimeException(); }
-    public  int productArity () { throw new RuntimeException(); }
-    public  Object productElement (int x$1) { throw new RuntimeException(); }
-    public  scala.collection.Iterator<java.lang.Object> productIterator () { throw new RuntimeException(); }
-    public  boolean canEqual (Object x$1) { throw new RuntimeException(); }
-    public  int hashCode () { throw new RuntimeException(); }
-    public  java.lang.String toString () { throw new RuntimeException(); }
-    private  java.lang.Object readResolve () { throw new RuntimeException(); }
+    public   A$ ()  { throw new RuntimeException(); }
+    public  java.lang.String productPrefix ()  { throw new RuntimeException(); }
+    public  int productArity ()  { throw new RuntimeException(); }
+    public  Object productElement (int x$1)  { throw new RuntimeException(); }
+    public  java.lang.String productElementName (int x$1)  { throw new RuntimeException(); }
+    public  scala.collection.Iterator<java.lang.Object> productIterator ()  { throw new RuntimeException(); }
+    public  boolean canEqual (Object x$1)  { throw new RuntimeException(); }
+    public  int hashCode ()  { throw new RuntimeException(); }
+    public  java.lang.String toString ()  { throw new RuntimeException(); }
+    private  java.lang.Object readResolve ()  { throw new RuntimeException(); }
   }
   static public  class B implements akka.rk.buh.is.it.Blarb.Fuz, scala.Product, scala.Serializable {
-    public  java.lang.String x () { throw new RuntimeException(); }
+    public  java.lang.String x ()  { throw new RuntimeException(); }
     // not preceding
-    public   B (java.lang.String x) { throw new RuntimeException(); }
-    public  akka.rk.buh.is.it.Blarb.B copy (java.lang.String x) { throw new RuntimeException(); }
-    public  java.lang.String copy$default$1 () { throw new RuntimeException(); }
+    public   B (java.lang.String x)  { throw new RuntimeException(); }
+    public  akka.rk.buh.is.it.Blarb.B copy (java.lang.String x)  { throw new RuntimeException(); }
+    public  java.lang.String copy$default$1 ()  { throw new RuntimeException(); }
     // not preceding
-    public  java.lang.String productPrefix () { throw new RuntimeException(); }
-    public  int productArity () { throw new RuntimeException(); }
-    public  Object productElement (int x$1) { throw new RuntimeException(); }
-    public  scala.collection.Iterator<java.lang.Object> productIterator () { throw new RuntimeException(); }
-    public  boolean canEqual (Object x$1) { throw new RuntimeException(); }
-    public  int hashCode () { throw new RuntimeException(); }
-    public  java.lang.String toString () { throw new RuntimeException(); }
-    public  boolean equals (Object x$1) { throw new RuntimeException(); }
+    public  java.lang.String productPrefix ()  { throw new RuntimeException(); }
+    public  int productArity ()  { throw new RuntimeException(); }
+    public  Object productElement (int x$1)  { throw new RuntimeException(); }
+    public  java.lang.String productElementName (int x$1)  { throw new RuntimeException(); }
+    public  scala.collection.Iterator<java.lang.Object> productIterator ()  { throw new RuntimeException(); }
+    public  boolean canEqual (Object x$1)  { throw new RuntimeException(); }
+    public  int hashCode ()  { throw new RuntimeException(); }
+    public  java.lang.String toString ()  { throw new RuntimeException(); }
+    public  boolean equals (Object x$1)  { throw new RuntimeException(); }
   }
-  // no position
   static public  class B$ extends scala.runtime.AbstractFunction1<java.lang.String, akka.rk.buh.is.it.Blarb.B> implements scala.Serializable {
     /**
      * Static reference to the singleton instance of this Scala object.
      */
     public static final B$ MODULE$ = null;
-    public   B$ () { throw new RuntimeException(); }
-    public final  java.lang.String toString () { throw new RuntimeException(); }
-    public  akka.rk.buh.is.it.Blarb.B apply (java.lang.String x) { throw new RuntimeException(); }
-    public  scala.Option<java.lang.String> unapply (akka.rk.buh.is.it.Blarb.B x$0) { throw new RuntimeException(); }
-    private  java.lang.Object readResolve () { throw new RuntimeException(); }
+    public   B$ ()  { throw new RuntimeException(); }
+    public final  java.lang.String toString ()  { throw new RuntimeException(); }
+    public  akka.rk.buh.is.it.Blarb.B apply (java.lang.String x)  { throw new RuntimeException(); }
+    public  scala.Option<java.lang.String> unapply (akka.rk.buh.is.it.Blarb.B x$0)  { throw new RuntimeException(); }
+    private  java.lang.Object readResolve ()  { throw new RuntimeException(); }
   }
   static public  interface Fuz {
   }
--- target/expected_output/basic/akka/rk/buh/is/it/CompressionProtocol.java
+++ target/expected_output/basic/akka/rk/buh/is/it/CompressionProtocol.java
@@ -9,6 +9,7 @@
       static public  akka.rk.buh.is.it.CompressionProtocol.Events.HeavyHitterDetected apply (Object key, int id, long count)  { throw new RuntimeException(); }
       static public  scala.Option<scala.Tuple3<java.lang.Object, java.lang.Object, java.lang.Object>> unapply (akka.rk.buh.is.it.CompressionProtocol.Events.HeavyHitterDetected x$0)  { throw new RuntimeException(); }
       static private  java.lang.Object readResolve ()  { throw new RuntimeException(); }
+      static public  scala.collection.Iterator<java.lang.String> productElementNames ()  { throw new RuntimeException(); }
       public  Object key ()  { throw new RuntimeException(); }
       public  int id ()  { throw new RuntimeException(); }
       public  long count ()  { throw new RuntimeException(); }
@@ -23,6 +24,7 @@
       public  java.lang.String productPrefix ()  { throw new RuntimeException(); }
       public  int productArity ()  { throw new RuntimeException(); }
       public  Object productElement (int x$1)  { throw new RuntimeException(); }
+      public  java.lang.String productElementName (int x$1)  { throw new RuntimeException(); }
       public  scala.collection.Iterator<java.lang.Object> productIterator ()  { throw new RuntimeException(); }
       public  boolean canEqual (Object x$1)  { throw new RuntimeException(); }
       public  int hashCode ()  { throw new RuntimeException(); }
--- target/expected_output/basic/akka/rk/buh/is/it/EWMA.java
+++ target/expected_output/basic/akka/rk/buh/is/it/EWMA.java
@@ -28,6 +28,7 @@
   static public  akka.rk.buh.is.it.EWMA apply (double value, double alpha) { throw new RuntimeException(); }
   static public  scala.Option<scala.Tuple2<java.lang.Object, java.lang.Object>> unapply (akka.rk.buh.is.it.EWMA x$0) { throw new RuntimeException(); }
   static private  java.lang.Object readResolve () { throw new RuntimeException(); }
+  static public  scala.collection.Iterator<java.lang.String> productElementNames ()  { throw new RuntimeException(); }
   public  double value () { throw new RuntimeException(); }
   public  double alpha () { throw new RuntimeException(); }
   // not preceding
@@ -48,6 +49,7 @@
   public  java.lang.String productPrefix () { throw new RuntimeException(); }
   public  int productArity () { throw new RuntimeException(); }
   public  Object productElement (int x$1) { throw new RuntimeException(); }
+  public  java.lang.String productElementName (int x$1)  { throw new RuntimeException(); }
   public  scala.collection.Iterator<java.lang.Object> productIterator () { throw new RuntimeException(); }
   public  boolean canEqual (Object x$1) { throw new RuntimeException(); }
   public  int hashCode () { throw new RuntimeException(); }
--- target/expected_output/basic/akka/rk/buh/is/it/Trait.java
+++ target/expected_output/basic/akka/rk/buh/is/it/Trait.java
@@ -6,6 +6,7 @@
     public  java.lang.String productPrefix () { throw new RuntimeException(); }
     public  int productArity () { throw new RuntimeException(); }
     public  Object productElement (int x$1) { throw new RuntimeException(); }
+    public  java.lang.String productElementName (int x$1)  { throw new RuntimeException(); }
     public  scala.collection.Iterator<java.lang.Object> productIterator () { throw new RuntimeException(); }
     public  boolean canEqual (Object x$1) { throw new RuntimeException(); }
     public  int hashCode () { throw new RuntimeException(); }
